addons/api/addon/transforms/object.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Transform from '@ember-data/serializer/transform';
+import { TrackedObject } from 'tracked-built-ins';
+import { typeOf } from '@ember/utils';
+
+export default class ObjectTransform extends Transform {
+  /**
+   * @param serialized
+   * @returns {object}
+   */
+  deserialize(serialized) {
+    if (typeOf(serialized) === 'object') {
```

Comment:
Why's this check needed? Shouldn't it always be an object or just undefined?

Reply:
yeah you are right, this is not needed I added this as more of a generic check, and now replaced this empty check